mongomock/collection.py

```diff
@@ -1570,8 +1570,13 @@ def _extend_collection(out_collection, field, expression):
                                     elif operator == "$last":
                                         doc_dict[field] = values[-1]
                                     elif operator == "$addToSet":
+                                        doc_dict[field] = []
```

Comment:
I suggest you use a local var to build the list and only assign it to `doc_dict[field]`. This will avoid always getting the field.

Reply:
I am not sure I understand exactly what you mean. Do you want me to update the rest of the code too (all previous elifs) ?

Reply:
No, just for yours:

```py
value = []
...
value.append(elt)
...
value.reverse()
...
doc_dict[field] = value
```